app/main.py

```diff
@@ -1 +1,32 @@
-# write your code here
+from __future__ import annotations
+
+
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.health = health
+        self.name = name
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (
+            f"{{Name: {self.name}, "
+            f"Health: {self.health}, "
+            f"Hidden: {self.hidden}}}"
+        )
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, obj: "Herbivore") -> None:
+        if isinstance(obj, Herbivore):
+            if not obj.hidden:
+                obj.health -= 50
+            if obj.health <= 0:
+                Animal.alive.remove(obj)
```

Comment:
You should use the class name to get access to the class attributes. So, this line should be `Herbivore.alive.remove(obj)`. This will make the code more efficient.